pkg/machines/libvirt-dbus.js

```diff
@@ -1392,6 +1402,14 @@ export function getAllInterfaces(dispatch, connectionName) {
             .fail(ex => console.warn('getAllInterfaces action failed:', JSON.stringify(ex)));
 }
 
+function initResource(connectionName, method, updateOrAddMethod, flags) {
+    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', method, [flags], TIMEOUT)
+            .then(objPaths => {
+                return Promise.all(objPaths[0].map(() => updateOrAddMethod({})));
```

Comment:
This could be one liner, and also the fail callback can be passed a second parameter to the then.
```
.then(objPaths => Promise.all(objPaths[0].map(() => updateOrAddMethod({})),
      ex => console.warn('initResource action failed:', JSON.stringify(ex))))
```